gpytorch/distributions/multivariate_normal.py

```diff
@@ -114,6 +120,18 @@ def get_base_samples(self, sample_shape=torch.Size()):
             base_samples = _standard_normal(shape, dtype=self.loc.dtype, device=self.loc.device)
         return base_samples
 
+    @property
+    def base_sample_shape(self):
+        """
+        Returns the shape of a base sample (without batching) that is used to
+        generate a single sample.
+        """
+        base_sample_shape = self.event_shape
+        if isinstance(self.lazy_covariance_matrix, RootLazyTensor):
+            base_sample_shape = self.lazy_covariance_matrix.root.shape[-1:]
```

Comment:
@wjmaddox `event_shape` for multitask MVNs will be a length-2 tuple. It might be worth overwriting this property on the MMVN class to explicitly return `self.event_shape`, just to be safe.

Reply:
hmm so what if the root decomp of the underlying full covariance of the MTMVN is not square? Shouldn't the handling here apply there as well? I guess it's not clear how a smaller/larger number of samples can be squeezed into the expected two-tuple form.

Reply:
I supoose it would only work in some sort of non-interleaved setting? But, ideally the interleaving? will cause the non-full rank LT to just not be a RootLazyTensor.